lib/routes/shared/security_pin/backup_phrase/generate_backup_phrase_page.dart

```diff
@@ -0,0 +1,143 @@
+import 'package:bip39/bip39.dart' as bip39;
+import 'package:breez/theme_data.dart' as theme;
+import 'package:breez/widgets/back_button.dart' as backBtn;
+import 'package:breez/widgets/route.dart';
+import 'package:flutter/material.dart';
+
+import 'verify_backup_phrase_page.dart';
+
+class GenerateBackupPhrasePage extends StatefulWidget {
+  GenerateBackupPhrasePage();
+
+  @override
+  GenerateBackupPhrasePageState createState() => new GenerateBackupPhrasePageState();
+}
+
+class GenerateBackupPhrasePageState extends State<GenerateBackupPhrasePage> {
+  String _mnemonics;
+  List<String> _mnemonicsList;
+
+  int _currentPage;
+
+  @override
+  void initState() {
+    _currentPage = 1;
+    _mnemonics = bip39.generateMnemonic(strength: 256);
+    _mnemonicsList = _mnemonics.split(" ");
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return WillPopScope(
+      onWillPop: () => _onWillPop(context),
+      child: Scaffold(
+        appBar: new AppBar(
+            iconTheme: theme.appBarIconTheme,
+            textTheme: theme.appBarTextTheme,
+            backgroundColor: theme.BreezColors.blue[500],
+            automaticallyImplyLeading: false,
+            leading: backBtn.BackButton(
+              onPressed: () => _onWillPop(context),
+            ),
+            title: new Text(
+              "Write these words ($_currentPage/2)",
+              style: theme.appBarTextStyle,
+            ),
+            elevation: 0.0),
+        body: _buildMnemonicSeedList(),
+        bottomNavigationBar: _buildNextBtn(),
+      ),
+    );
+  }
+
+  _onWillPop(BuildContext context) {
```

Comment:
Isn't it better to push each page to the stack and let the back button and native back behaves naturally?